File: solutions/problem_0065_valid_number/solution.py
"""
Problem 65: Valid Number
========================
Difficulty: Hard
Tags: String
Pattern: String Parsing / State Machine

Time Complexity:  O(n)
Space Complexity: O(1)
"""

class Solution:
    def isNumber(self, s: str) -> bool:
        try:
            float(s)
            return True
        except ValueError:
            return False

File: solutions/problem_0065_valid_number/solution.cpp
/*
 * Problem 65: Valid Number
 * ========================
 * Difficulty: Hard
 * Tags: String
 * Pattern: String Parsing / State Machine
 *
 * Time Complexity:  O(n)
 * Space Complexity: O(1)
 */

#include <algorithm>
#include <cctype>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    bool isNumber(string& s) {
        // String processing approach - O(n) time
        string processed;
        for (char ch : s) {
            if (isalnum(ch)) {
                processed += tolower(ch);
            }
        }
        string rev = processed;
        reverse(rev.begin(), rev.end());
        return processed == rev;
    }
};
